identity-account-storage/src/types/signature.rs

```diff
@@ -1,33 +1,20 @@
 // Copyright 2020-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use identity_core::crypto::PublicKey;
-#[cfg(feature = "bindings-derive")]
 use serde::Deserialize;
-#[cfg(feature = "bindings-derive")]
 use serde::Serialize;
 
-/// A digital signature and associated public key.
-#[cfg_attr(feature = "bindings-derive", derive(Clone, Deserialize, Serialize))]
-#[cfg_attr(not(feature = "bindings-derive"), derive(Clone))]
-pub struct Signature {
-  pub(crate) pkey: PublicKey,
-  pub(crate) data: Vec<u8>,
+/// A digital signature.
+#[derive(Clone, Deserialize, Serialize)]
+pub enum Signature {
+  Ed25519(Vec<u8>),
```

Comment:
I think this design is good for forwards-compatibility. Should we mark the enum as `non_exhaustive` so we can add variants without causing a breaking change?